dbms/src/Common/ZooKeeper/ZooKeeper.h

```diff
@@ -161,9 +160,11 @@ class ZooKeeper
     int32_t tryRemoveEphemeralNodeWithRetries(const std::string & path, int32_t version = -1, size_t * attempt = nullptr);
 
     bool exists(const std::string & path, Stat * stat = nullptr, const EventPtr & watch = nullptr);
+    bool exists(const std::string & path, Stat * stat, const WatchCallback & watch_callback);
```

Comment:
There is no need for this method as it is the same as existsWatch(). The reason for the -Watch suffix is that overload resolution becomes somewhat difficult because there is the other method with a lot of default parameters.

Anyway, the interface for get(), tryGet() and exists() should be consistent. So you probably should add and use the getWatch() method, delete this exists() overload and use existsWatch() method.

Reply:
Fixed in 4361df913b126bf89c41c11a1058464faad75ede . Here we need to take into account that libzookeper will be removed soon, and the new implementation that @alexey-milovidov have a slightly different interface. We will see what conflicts we will have on merge.

Reply:
Also for consistency getWatch() instead of get().

Yes, there will be some conflicts. Who knows which pull will get merged first :)

Reply:
Fixed in 438121e45b5908f783b1fdc09a8964dc63617dc3

Reply:
👍